src/langcheck/metrics/en/reference_based_text_quality.py

```diff
@@ -71,15 +72,24 @@ def semantic_similarity(
         'local', 'openai', 'azure_openai'
     ], ('Unsupported embedding model type. '
         'The supported ones are ["local", "openai", "azure_openai"]')
-
+    batch_size = 8
     if model_type == 'local':
         # The 'all-mpnet-base-v2' model has the highest average performance out
         # of all the existing sentence-transformer models that have been
         # evaluated.
         # Ref: https://www.sbert.net/docs/pretrained_models.html#model-overview
         model = SentenceTransformer('sentence-transformers/all-mpnet-base-v2')
-        generated_embeddings = model.encode(generated_outputs)
-        reference_embeddings = model.encode(reference_outputs)
+        generated_embeddings = []
+        reference_embeddings = []
+        for i in tqdm_wrapper(range(0, len(generated_outputs), batch_size),
+                              total=len(generated_outputs) // batch_size,
```

Comment:
`total` should be `0` when `len(generated_output) == 0`, and `1` when `len(generated_output) == 1`.
```suggestion
                              total=(len(generated_outputs) + batch_size - 1)// batch_size,
```